demo_modules/slideshow/slideshow.js

```diff
@@ -0,0 +1,336 @@
+      });
+  }
+}
+
+class StaticServerDisplayStrategy extends ServerDisplayStrategy {
+  constructor(config) {
+    super();
+    this.config = config;
+    
+    // Keep track of content indices. When our content array's length doesn't
+    // match the length of this array, add the new indices to this array.
+    this.nextContentIndices = [];
+    
+    // The time we last updated a display.
+    this.lastUpdate = 0;
+    
+    let contentHasArrived = new Promise((resolve, reject) => {
+      this.contentArrives = resolve;
+    });
+    network.on('connection', (socket) => {
+      contentHasArrived.then((content) => {
+        this.chooseSomeContent(content, socket);
+        return content;
+      });
+    });
+  }
+  init() {
+    // Return a promise when initialization is complete.
+    return Promise.resolve();
+  }
+  chooseSomeContent(content, socket) {
+    assert(this.nextContentIndices.length, 'No content to select from!');
+    // Choose the next one.
+    let index = this.nextContentIndices.shift();
+    
+    debug('Sending content index ' + index + ' to client.');
+    
+    // Send it to the specified client.
+    socket.emit('content', content[index]);
+    // Add this index back to the end of the list of indices.
+    this.nextContentIndices.push(index);
+  }
+  tick(content, time, delta) {
+    // If there's no content to show, just stop.
+    if (!content || !content.length) {
+      return;
+    }
+    
+    assert(content.length >= this.nextContentIndices.length,
+           'Whoa! Weird edge case: We never expect to unload any content!',
+           content.length, this.nextContentIndices.length);
+           
+    if (content.length > this.nextContentIndices.length) {
+      // We've loaded new content. Generate a list of new indices and shuffle.
+      let newIndices = _.shuffle(_.range(this.nextContentIndices.length, content.length));
+      // Add the content indices.
+      this.nextContentIndices.push(...newIndices);
+    }
+    
+    this.contentArrives(content);
+    
+    // Otherwise, tell a specific client to show a specific bit of content.
+    if (time - this.lastUpdate >= this.config.period) {
+      // Pick a random client.
+      let client = _.sample(network.getClientsInRect(wallGeometry.extents));
+      if (client) {
+        this.chooseSomeContent(content, client.socket);
+      }
+      this.lastUpdate = time;
+    }
+  }
+  serializeForClient() {
+    return {'static': this.config};
+  }
+}
+
+class StaticClientDisplayStrategy extends ClientDisplayStrategy {
+  display(container, loadedContent) {
+    // Assume loadedContent is an element...
+    while (container.firstChild) {
+      container.removeChild(container.firstChild);
+    }
+    container.appendChild(loadedContent);
+  }
+}
+
+class ImageServer extends ServerModuleInterface {
+  constructor(config) {
+    super();
+    
+    // The load strategy for this run of the module.
+    this.loadStrategy = parseServerLoadStrategy(config.load);
```

Comment:
You're also not handling the null strategies here.  Maybe that's fine and instead you should have parse* throw with a useful error message instead of generating that error message everywhere where it is called.